tools/crm_ticket.c

```diff
@@ -736,21 +746,34 @@ main(int argc, char **argv)
     attr_delete = NULL;
 
     args = pcmk__new_common_args(SUMMARY);
-    context = build_arg_context(args);
+    context = build_arg_context(args, &output_group);
     processed_args = pcmk__cmdline_preproc(argv, "dintvxCDGS");
 
+    pcmk__register_formats(output_group, formats);
     if (!g_option_context_parse_strv(context, &processed_args, &error)) {
         exit_code = CRM_EX_USAGE;
         goto done;
     }
 
     pcmk__cli_init_logging("crm_ticket", args->verbosity);
 
+    rc = pcmk__output_new(&out, args->output_ty, args->output_dest, argv);
+    if (rc != pcmk_rc_ok) {
+        exit_code = pcmk_rc2exitc(rc);
+        g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
+                    "Error creating output format %s: %s", args->output_ty,
+                    pcmk_rc_str(rc));
+        goto done;
+    }
+
+    if (!pcmk__force_args(context, &error, "%s --xml-simple-list", g_get_prgname())) {
```

Comment:
No, there isn't.  Removing this will change the list output to this style:
```
<list name="tickets" count="2">
   <ticket ...>
   <ticket ...>
</list>
```

I no longer remember why we do one over the other.  The non-simple list style seems less XMLish to me anymore, but we've got a mix of programs using each way.

Reply:
IIRC we added simple list to maintain previous behavior somewhere, and I guess it just sort of grew from there. I don't have a preference one way or the other as to the style, but I suspect we have the problem where the libpacemaker API output won't follow the schema used by the tools. Maybe we should convert it to a begin_list() argument

Reply:
It looks like this (and --xml-legacy) were added to preserve the --as-xml behavior in crm_mon.  It is also currently in use in pacemakerd, crmadmin, crm_node, crm_simulate, and crm_resource.  We probably shouldn't have used it in those other tools, and I assume it's too late to start taking it back out.

I also think you're right that libpacemaker won't follow the same schema because of this - I don't see us forcing the argument anywhere in the library.  It's definitely time to start adding further tests here.

I guess I'm not quite sure how to start cleaning up this mess.

Reply:
But I think it's worth revisiting one of my previous comments - the new <list> lists don't really strike me as especially XMLish.  The simple list output looks more like what I would expect if I were just looking at a random XML blob.  I don't remember why we thought that was a good idea.

Reply:
> It looks like this (and --xml-legacy) were added to preserve the --as-xml behavior in crm_mon. It is also currently in use in pacemakerd, crmadmin, crm_node, crm_simulate, and crm_resource. We probably shouldn't have used it in those other tools, and I assume it's too late to start taking it back out.

Yeah ...
 
> I also think you're right that libpacemaker won't follow the same schema because of this - I don't see us forcing the argument anywhere in the library. It's definitely time to start adding further tests here.
> 
> I guess I'm not quite sure how to start cleaning up this mess.

We should probably drop the command-line option. We could make it a new global, or a begin_list() argument, or maybe something like substitutions[]

> But I think it's worth revisiting one of my previous comments - the new lists don't really strike me as especially XMLish. The simple list output looks more like what I would expect if I were just looking at a random XML blob. I don't remember why we thought that was a good idea.

I don't have a preference. I believe the original idea was that automated processing of the XML would be easier if it reused schema elements as much as possible.

Reply:
In trying to decide what to do here, I've noticed that we basically do three different things:

```
<list name="things" count="3">
   <item name="thing" />
   <item name="thing" />
   <item name="thing" />
</list>
```

I think only stonith_admin does that.

```
<things>
   <thing />
   <thing />
   <thing />
</things>
```

Lots of places do this.

```
<pacemaker-result>
   <thing />
   <thing />
   <thing />
   <status code="" message="" />
</pacemaker-result>
```

A surprisingly large number of places do this.

Reply:
Rebasing on #3365 takes care of this.